File: 002CustomManagedObject_OSX_ObjC/002CustomManagedObject_OSX_ObjC/ViewController.m
//
//  ViewController.m
//  002CustomManagedObject_OSX_ObjC
//
//  Created by Henry AT on 6/19/15.
//  Copyright (c) 2015 Apps4s. All rights reserved.
//

#import "ViewController.h"

@implementation ViewController

- (void)viewDidLoad {
  [super viewDidLoad];

  // Do any additional setup after loading the view.
}

- (void)setRepresentedObject:(id)representedObject {
  [super setRepresentedObject:representedObject];

  // Update the view, if already loaded.
}

- (IBAction)btnCreatedCustomManagedObject:(id)sender {
  
  AppDelegate *appDelegate = (AppDelegate *)[[NSApplication sharedApplication] delegate];
  NSManagedObjectContext *context = appDelegate.managedObjectContext;
  
  //  NSManagedObject *course = [NSEntityDescription insertNewObjectForEntityForName:@"Course" inManagedObjectContext:context];
  Course *course = (Course *)[NSEntityDescription insertNewObjectForEntityForName:@"Course" inManagedObjectContext:context];
   
  course.title = @"Curso de CoreData for iOS and OS X con Objetive-C";
  course.author = @"Henry AT";
  course.releaseDate = [NSDate date];
  
  [course simpleMethod];
  // Ahora grabaremos, para no tener la necesidad de salir del Simulador
  
  NSError *error = nil;
  
  if (![context save:&error]) {
    NSLog(@"An Error! %@",error);
  } else{
    NSLog(@"Todo se grabo OK..");
  }
  
  
  
}
@end

File: 002CustomManagedObject_OSX_Swift/002CustomManagedObject_OSX_Swift/ViewController.swift
//
//  ViewController.swift
//  002CustomManagedObject_OSX_Swift
//
//  Created by Henry AT on 6/19/15.
//  Copyright (c) 2015 Apps4s. All rights reserved.
//

import Cocoa
import CoreData



class ViewController: NSViewController {

  override func viewDidLoad() {
    super.viewDidLoad()

    // Do any additional setup after loading the view.
  }

  override var representedObject: AnyObject? {
    didSet {
    // Update the view, if already loaded.
    }
  }


  
  @IBAction func btnCreatedCustomManagedObject(sender: AnyObject) {
    
    
    let appDelegate = NSApplication.sharedApplication().delegate as! AppDelegate
    var context: NSManagedObjectContext? = appDelegate.managedObjectContext
    
    
    let course = NSEntityDescription.insertNewObjectForEntityForName("Course", inManagedObjectContext: context!) as! Course
    
    course.title = "Curso de CoreData for iOS and OS X en Swift"
    course.author = "Henry"
    course.releaseDate = NSDate()
    
    course.simpleMethod()
    // Ahora grabaremos, para no tener la necesidad de salir del Simulador
    var error: NSError?
    
    if((context?.save(&error)) != nil){
      println("Todo se grabo OK..")
    }else{
      println("An Error! \(error!.localizedDescription)")
    }
    
    
    
  }
}

